lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2610,23 +2629,55 @@ namespace {
         }
       }
 
+      auto contextIsEnum = [&](const ImportedName &name) -> bool {
+        EffectiveClangContext importContext = name.getEffectiveContext();
+        switch (importContext.getKind()) {
+        case EffectiveClangContext::DeclContext:
+          return importContext.getAsDeclContext() == canonicalClangDecl;
+        case EffectiveClangContext::TypedefContext: {
+          auto *typedefName = importContext.getTypedefName();
+          clang::QualType underlyingTy = typedefName->getUnderlyingType();
+          return underlyingTy->getAsTagDecl() == canonicalClangDecl;
+        }
+        case EffectiveClangContext::UnresolvedContext:
+          llvm_unreachable("can't import into an unresolved context");
```

Comment:
I think the only way we could get here is if someone renamed an enumerator onto *another* decl, which won't work. But we shouldn't crash. I'll change it to just return false.